tests/v2/integration/projects/resource_quota_test.go

```diff
@@ -203,16 +206,35 @@ func (s *ResourceQuotaSuite) TestRemoveQuotaFromProjectWithNamespacePropagation(
 	err = steveResourceQuotas.CheckResourceActiveState(client, quotaID)
 	s.Require().NoError(err)
 
-	quotas, err = resourcequotas.ListResourceQuotas(client, localClusterID, resourceQuotaNamespaceName, metav1.ListOptions{})
-	s.Require().NoError(err)
-	s.Require().NotNil(quotas)
-	s.Require().Lenf(quotas.Items, 1, "Expected 1 quota in the namespace, but got %d", len(quotas.Items))
-
-	resourceList := quotas.Items[0].Spec.Hard
+	// Wait a little while for the resourcequota to be updated.
+	// The quota controller sometimes gets conflict error trying to update the namespace annotation and needs time to retry.
 	want := v1.ResourceList{
 		v1.ResourceConfigMaps: resource.MustParse("5"),
 	}
-	s.Require().Equal(want, resourceList, "Expected the CPU limits to be removed, but config maps limit to remain")
+	var resourceList v1.ResourceList
+	err = kwait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+		quotas, err = resourcequotas.ListResourceQuotas(client, localClusterID, resourceQuotaNamespaceName, metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		if quotas == nil {
+			return false, nil
+		}
+		if len(quotas.Items) != 1 {
+			return false, nil
```

Comment:
I'm not sure why this is different from line 331. Why would it not be an error here but an error at line 331?